### PR TITLE
Fix pattern translation for SP escapes, slash and empty pattern

### PR DESCRIPTION
**Pull request by Exter-N**
Fixes calling the constructor with Supplementary Plane escapes (like `new RE2("\\u{1F603}")`) (see also #27).

Also adds some tests for Unicode Supplementary Plane related stuff (see also #21), and adds a missing bound check in the C++ land.

Also changes the pattern translation for `/` and the empty pattern to follow the ES2015 specification (see also #24).

### CODE REVIEW

File: lib/new.cc [hunk 1]
```diff
@@ -191,7 +215,7 @@ NAN_METHOD(WrappedRE2::New) {
 		return Nan::ThrowTypeError("Expected string, Buffer, RegExp, or RE2 as the 1st argument.");
 	}
 
-	if (needConversion && data && size && translateRegExp(data, size, buffer)) {
+	if (needConversion && translateRegExp(data, size, buffer)) {
```
Comment (Exter-N) [inline lib/new.cc]: I removed the check for `data` here because it is unnecessary (checked 4 lines earlier), and the check for `size` because `translateRegExp` now has something to do when `!size`.
